macOS/DuckDuckGo/DefaultBrowserAndAddToDockPrompts/DefaultBrowserAndDockPromptPresenting.swift

```diff
@@ -0,0 +1,180 @@
+            return nil
+        }
+
+        let content = DefaultBrowserAndDockPromptContent.banner(type)
+
+        /// We mark the banner as shown when it gets actioned (eithe dismiss or confirmation)
+        /// Given that we want to show the banner in all windows.
+        return BannerMessageViewController(
+            message: content.message,
+            image: content.icon,
+            buttonText: content.primaryButtonTitle,
+            buttonAction: {
+                self.coordinator.onPromptConfirmation()
+                self.repository.setPromptShown(true)
+                self.bannerDismissedSubject.send()
+            },
+            closeAction: {
+                self.bannerDismissedSubject.send()
+                self.repository.setPromptShown(true)
+            })
+    }
+
+    private func createPopover(with type: DefaultBrowserAndDockPromptType) -> NSHostingController<DefaultBrowserAndDockPromptPopoverView> {
+        let content = DefaultBrowserAndDockPromptContent.popover(type)
+        let viewModel = DefaultBrowserAndDockPromptPopoverViewModel(
+            title: content.title,
+            message: content.message,
+            image: content.icon,
+            buttonText: content.primaryButtonTitle,
+            buttonAction: {
+                self.coordinator.onPromptConfirmation()
+                self.popover?.close()
+            },
+            secondaryButtonText: content.secondaryButtonTitle,
+            secondaryButtonAction: {
+                self.popover?.close()
+            })
+
+        let contentView = DefaultBrowserAndDockPromptPopoverView(viewModel: viewModel)
+
+        return NSHostingController(rootView: contentView)
+    }
+
+    private func initializePopover(with type: DefaultBrowserAndDockPromptType) {
+        let viewController = createPopover(with: type)
+        popover = DefaultBrowserAndDockPromptPopover(viewController: viewController)
+    }
+
+    private func showPopover(positionedBelow view: NSView) {
+        popover?.show(positionedBelow: view)
+        popover?.contentViewController?.view.makeMeFirstResponder()
+    }
+
+    private func subscribeToLocalOverride() {
+        guard let overridesHandler = featureFlagger.localOverrides?.actionHandler as? FeatureFlagOverridesPublishingHandler<FeatureFlag> else {
+            return
+        }
+
+        overridesHandler.experimentFlagDidChangePublisher
+            .filter { $0.0 == .popoverVsBannerExperiment }
+            .sink { (_, cohort) in
+                if FeatureFlag.PopoverVSBannerExperimentCohort.cohort(for: cohort) == nil { return }
+
+                /// For testing purposes when we override the local features and because we want to show the prompt.
+                /// We set the set prompt flag to false in case it was show in the past.
+                self.repository.setPromptShown(false)
+
+                NotificationCenter.default.post(name: .showPromptForSetAsDefaultBrowserAndAddToDock, object: nil)
```

Comment:
It looks like this is only used to support the local overrides? This is fine, just wanted to make sure.

Overall I believe notifications should be called based on what happened and not on how it should be handled - if that makes sense. So I’d call it e.g. `.setAsDefaultBrowserAndAddToDockExperimentFlagOverrideDidChange`.